internal/log/base_handler.go

```diff
@@ -17,30 +19,44 @@ var stringMap = [...]string{
 }
 
 type baseHandler struct {
-	mu     sync.Mutex
-	Writer io.Writer
+	mutex   sync.Mutex
+	test    bool
+	Writer  io.Writer
+	Entries []string
 }
 
-func newBaseHandler(w io.Writer) *baseHandler {
+func newBaseHandler(w io.Writer) (*baseHandler, error) {
 	return &baseHandler{
 		Writer: w,
+	}, nil
+}
+
+func baselFields(e *apexLog.Entry) map[string]string {
+	out := make(map[string]string)
+
+	for key := range e.Fields {
+		val := fmt.Sprint(e.Fields[key])
+		out[key] = val
 	}
+
+	return out
 }
 
 func (h *baseHandler) HandleLog(e *apexLog.Entry) error {
-	level := stringMap[e.Level]
-	names := e.Fields.Names()
+	bld := strings.Builder{}
+	bld.WriteString(e.Message + " ")
 
-	h.mu.Lock()
-	defer h.mu.Unlock()
+	enc := json.NewEncoder(&bld)
+	enc.Encode(baselFields(e))
```

Comment:
enc.Encode adds a newline character, but so does fmt.Fprintln a few lines later.  This means every log message now ends in *two* newline characters.  I think the Fprintln should be changed to just Fprintf to fix this.